src/utils/usecase.ts

```diff
@@ -70,6 +72,15 @@ export const getTransactionMethods = <
     args.pop();
   }
 
+  const buildTransaction = async (overrides?: Overrides) => {
+    const mergedOverrides = { ...initialOverrides, ...overrides };
+    const populatedTransaction = await contract.populateTransaction[
+      method as string
+    ](...[...args, mergedOverrides]);
+    populatedTransaction.data = populatedTransaction.data + tag;
```

Comment:
Should the tag be created here (with the domain being passed into `getTransactionMethods`)?

Reply:
that would dedupe more of the `getTagFromDomain` uses that are passed in so I don't see why not. also `domain` seems more self explanatory than `tag` so that would be another benefit.